### PR TITLE
getGeneratorNames added

### PR DESCRIPTION
**Pull request by twixie**
added GetGeneratorNames. First git pull request! :tada

### CODE REVIEW

**Comment by twixie** [comment]
19

File: package.json [hunk 1]
```diff
@@ -44,7 +44,8 @@
     "log-symbols": "^1.0.1",
     "mem-fs": "^1.1.0",
     "text-table": "^0.2.0",
-    "untildify": "^2.0.0"
+    "untildify": "^2.0.0",
+    "array-uniq": "~1.0.0"
```
Comment (sindresorhus) [inline package.json]: use `^`

File: lib/environment.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ var debug = require('debug')('generators:environment');
 var Store = require('./store');
 var resolver = require('./resolver');
 var TerminalAdapter = require('./adapter');
+var arrayUniq = require("array-uniq");
```
Comment (sindresorhus) [inline lib/environment.js]: single-quotes

File: lib/environment.js [hunk 3]
```diff
@@ -208,6 +209,16 @@ Environment.prototype.getGeneratorsMeta = function getGeneratorsMeta() {
 };
 
 /**
+ * Get registered generators names
+ * 
+ * @return {Array}
+ */
+
+Environment.prototype.getGeneratorNames = function getGeneratorNames() {
+  arrayUniq(Object.keys(this.getGeneratorsMeta()).map(Generator.namespaceToName));
```
Comment (sindresorhus) [inline lib/environment.js]: missing a return

**Comment by sindresorhus** [comment]
Congrats on first PR \o/

Can you add a test for this?

File: package.json (hunk 1 above)
Comment (SBoudrias) [inline package.json]: Also, we already uses lodash. Use `_.uniq` instead of adding a new dependency.

**Comment by sindresorhus** [comment]
ping @twixie

**Comment by twixie** [comment]
I will update the PR but have troubles with my local github app. Give me some hours to figure it out. Thanks for ping me.

**Comment by arthurvr** [comment]
@twixie Would you be able to look into finishing this up? Thanks.

**Comment by twixie** [comment]
updates done. i don't really know qunit but will look into adding testing soon

**Comment by arthurvr** [comment]
> i don't really know qunit but will look into adding testing soon

We use Mocha to test our modules :smile:

**Comment by twixie** [comment]
Your tests are timing out! This is **not** my error!!!! https://github.com/yeoman/environment/pull/42 did it too!!! I didn't break anything!!!!

**Comment by SBoudrias** [comment]
@twixie You hit an issue with Travis.CI - but it's your code that is broke. See this log https://travis-ci.org/yeoman/environment/jobs/70384541

**Comment by twixie** [comment]
Please merge. I fixed ci already a few hours ago!

File: test/environment.js [hunk 4]
```diff
@@ -338,6 +338,18 @@ describe('Environment', function () {
...
+  
+  describe('#getGeneratorNames', function () {
+    beforeEach(function () {
+      this.generatorPath = path.join(__dirname, './fixtures/custom-generator-simple');
+      this.env.register(this.generatorPath);
+    });
+    
+    it('get the registered generators names', function () {
+      var names = this.env.getGeneratorNames();
+      assert(Array.isArray(names));
```
Comment (arthurvr) [inline test/environment.js]: Let's test what this actually returns too :)

**Comment by twixie** [comment]
done i will also test the namespace to name convertion

**Comment by twixie** [comment]
no that is already tested sorry! please merge

**Comment by twixie** [comment]
@sindresorhus @SBoudrias @arthurvr Please merge.

**Comment by SBoudrias** [comment]
Merged on master.

Next time, it's better to squash your commits before - also careful with the trailing whitespaces :)

Thanks for the hard work.

**Pull request closed without merging**